src/mxfeatures.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file mxfeatures.cc
+ * \brief check MXNet features including compile time support
+ */
+
+#include "mxnet/mxfeatures.h"
+#include <bitset>
+
+namespace mxnet {
+namespace features {
+
+class FeatureSet {
+ public:
+  FeatureSet() :
+      feature_bits() {
+    // GPU
+    feature_bits.set(CUDA, MXNET_USE_CUDA);
+    feature_bits.set(CUDNN, MXNET_USE_CUDNN);
+    feature_bits.set(NCCL, MXNET_USE_NCCL);
+    feature_bits.set(CUDA_RTC, MXNET_ENABLE_CUDA_RTC);
+    feature_bits.set(TENSORRT, MXNET_USE_TENSORRT);
+
+    // Check flags for example with gcc -msse3 -mavx2 -dM -E - < /dev/null | egrep "SSE|AVX"
+#if __SSE__
```

Comment:
```suggestion
feature_bits.set(CUDA, bool(__SSE__))
```
for the sake of consistency?

Reply:
Can't It's a macro defined only if present

Reply:
Resolving